scripts/include/helpers.inc.php

```diff
@@ -121,45 +121,98 @@ function mergeXliffDomToAtom(DOMDocument $domA, DOMDocument $domB): DOMDocument
 
 function mergeXliffDomToWeblate(DOMDocument $domA, DOMDocument $domB): DOMDocument
 {
-  // Elements in A but not in B -> copy to B
-  $sourcesA = $domA->getElementsByTagName('source');
-  $sourcesB = $domB->getElementsByTagName('source');
+  foreach ($domB->getElementsByTagName('trans-unit') as $element) {
+    if ($element->hasAttribute('id')) {
+        $element->setIdAttribute('id', true);
+    }
+  }
+
+  // Elements in A AND B -> check source string and if different -> replace in B
+  $transUnitsA = $domA->getElementsByTagName('trans-unit');
+  $transUnitsB = $domB->getElementsByTagName('trans-unit');
+
+  $i = 0;
+  foreach ($transUnitsA as $transUnit) {
+    $idListA[] = $transUnitsA->item($i)->getAttribute('id');
+    $i++;
+  }
 
-  foreach ($sourcesA as $source) {
-    $stringsA[] = $source->nodeValue;
+  $i = 0;
+  foreach ($transUnitsB as $transUnit) {
```

Comment:
Suggestion: The foreach loop at lines 124-128 could be combined with this one too I think?

Reply:
Done.